uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -186,12 +188,21 @@ sub determineSubjectID {
 	    exit 2;
         }
     }
+
+    # Check for regular expression pattenr in the identifier field
+    if ($tarchiveInfo->{'PatientName'} =~ /$Settings::regex_pattern/i){
+          $identifier = $tarchiveInfo->{'PatientName'};
+    }
+    elsif ($tarchiveInfo->{'PatientID'} =~ /$Settings::regex_pattern/i){
+          $identifier = $tarchiveInfo->{'PatientID'};
+    }
+
     my $subjectIDsref = Settings::getSubjectIDs(
-                            $tarchiveInfo->{'PatientName'},
-                            $tarchiveInfo->{'PatientID'},
-                            $scannerID,
-                            $this->{dbhr}
-                        );
+                                $identifier,
+                                $scannerID,
+                                $this->{dbhr}
+                            );
+
```

Comment:
@stronnics,
this function has been updated to take 3 arguments ($identifier now replaces PatientName and PatientID), but your dicom-archive-tools pull request that updated the profileTemplate file did not update this function (it still takes in 4 arguments)